src/Context/CartContext.jsx

```diff
@@ -1,9 +1,16 @@
-import { createContext, useContext, useState } from "react";
+import { createContext, useContext, useState, useEffect } from "react";
 
 const cartContext = createContext([]);
 
 const CartProvider = ({ children }) => {
-  const [cart, setCart] = useState([]);
+  const [cart, setCart] = useState(
+    JSON.parse(sessionStorage.getItem("cart")) ?? []
```

Comment:
Maybe make a util function, which take ***key*** and ***data** as arguments and do this logic there. This way, you'll remove the redundant checks and parsing every single time, you use sessionStorage.

Reply:
Yes, that would be a great solution. Thank you for the suggestion.